src/utils.ts

```diff
@@ -289,6 +286,8 @@ export async function askToConfigureEnvironment(
 export async function askAndOpenProject(
```

Comment:
> askAndOpenProject [](start = 21, length = 18)

The processing logic in function 'askAndOpenProject' and 'askAndOpenInRemote' looks very similar, except that one handles iot projects and the other handles containerized projects. Suggest abstract and refine them.